src/app/modules/trade-companion/trade-companion.module.ts

```diff
@@ -0,0 +1,97 @@
+import { TradeCompanionUserSettings } from '@shared/module/poe/type/trade-companion.type'
+import { SharedModule } from '@shared/shared.module'
+import { UserSettingsFeature } from 'src/app/layout/type'
+import { TradeCompanionStashGridComponent } from './component/stash-grid/trade-companion-stash-grid.component'
+import { TradeCompanionSettingsComponent } from './component/trade-companion-settings/trade-companion-settings.component'
+import { TradeNotificationComponent } from './component/trade-notification/trade-notification.component'
+import { TradeNotificationPanelComponent } from './component/trade-notifications-panel/trade-notifications-panel.component'
+
+@NgModule({
+  providers: [{ provide: FEATURE_MODULES, useClass: TradeCompanionModule, multi: true }],
+  declarations: [
+    TradeCompanionSettingsComponent,
+    TradeCompanionStashGridComponent,
+    TradeNotificationComponent,
+    TradeNotificationPanelComponent,
+  ],
+  imports: [SharedModule],
+  exports: [
+    TradeCompanionStashGridComponent,
+    TradeNotificationPanelComponent
+  ],
+})
+export class TradeCompanionModule implements FeatureModule {
+  constructor() { }
+
+  public getSettings(): UserSettingsFeature {
+    const maxVisibileTradeNotifications = 8
+    const defaultSettings: TradeCompanionUserSettings = {
+      tradeCompanionEnabled: false,
+      tradeCompanionOpacity: 1.0,
+      maxVisibileTradeNotifications: maxVisibileTradeNotifications,
+      incomingTradeOptions: [
+        {
+          buttonLabel: '1m',
+          whisperMessage: '1 minute please.',
+          kickAfterWhisper: false,
+          dismissNotification: false,
+        },
+        {
+          buttonLabel: 'thx',
+          whisperMessage: 'Thank you very much.',
+          kickAfterWhisper: true,
+          dismissNotification: true,
+        },
+        {
+          buttonLabel: 'sold',
+          whisperMessage: 'Sorry, already sold.',
+          kickAfterWhisper: true,
+          dismissNotification: true,
+        },
+      ],
+      outgoingTradeOptions: [
+        {
+          buttonLabel: 'thx',
+          whisperMessage: 'Thank you very much.',
+          kickAfterWhisper: true,
+          dismissNotification: true,
+        },
+      ],
+      stashGridBounds:
+        [
+          {
+            x: 16,
+            y: 134,
+            width: 624,//12*52px
+            height: 624,
+          },
+          {
+            x: 16,
+            y: 134,
+            width: 624,//24*26px
+            height: 624,
+          }
+        ],
+      stashGrids: new Map(),
+      showStashGridOnTrade: true,
+    }
+    return {
+      name: 'trade-companion.name',
+      component: TradeCompanionSettingsComponent,
+      defaultSettings,
+    }
+  }
+
+  public getFeatures(settings: TradeCompanionUserSettings): Feature[] {
+    return [{
+      name: 'stash-grid',
+      accelerator: undefined,
+    }]
+  }
+
+  public run(feature: string, settings: TradeCompanionUserSettings): void {
```

Comment:
another empty function

Reply:
Has to be around (due to interface implementation), so I've added a comment to it. (commit: 5b14e87)